rustls/src/client/hs.rs

```diff
@@ -720,6 +782,21 @@ impl State<ClientConnectionData> for ExpectServerHello {
             ));
         }
 
+        cx.common.negotiated_version = Some(version);
+        // Validate server and client cert type extension
+        if !cx.common.is_tls13() {
```

Comment:
It seems a little bit odd that there's a branching here based on protocol version. Can you expand on this design choice?

Reply:
Reason for this is that we want to fail if in the `client_cert_type` or `server_cert_type` extension both a x509 and RPK are offered. However, in tls13 the extension is encrypted so we call `process_client_server/cert_type_extension()` in `tls13::ExpectEncryptedExtensions`. To prevent calling this function twice we added the branching.

Reply:
Makes sense :+1: It sounds like consensus is moving towards not implementing this for TLS1.2, but if support stays it would be useful to capture this context at the call-site with a comment (IMO).